src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,595 @@
+
+            headerNodes[3].NameNode.Text.Should().Be("ContentLength");
+            headerNodes[3].SeparatorNode.Text.Should().Be(":");
+            headerNodes[3].ValueNode.Text.Should().Be("7060");
+
+            headerNodes[4].NameNode.Text.Should().Be("Cookie");
+            headerNodes[4].SeparatorNode.Text.Should().Be(":");
+            headerNodes[4].ValueNode.Text.Should().Be("expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;");
+        }
+    }
+
+    public class Body
+    {
+        [Fact]
+        public void header_with_body_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+                POST https://example.com/comments HTTP/1.1
+                Content-Type: application/xml
+                Authorization: token xxx
+
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+
+            using var _ = new AssertionScope();
+
+            var requestNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.HeaderNodes.Count.Should().Be(2);            
+            requestNode.BodyNode.Text.Should().Be(
+                """
+                <request>
+                    <name>sample</name>
+                    <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+                </request>
+                """);
+        }
+
+        [Fact]
+        public void header_separator_is_present()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+                POST https://example.com/comments HTTP/1.1
+                Content-Type: application                                                                                                            
+                """);         
+
+            var requestNode = result.SyntaxTree.RootNode
+                    .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.HeadersNode.ChildNodes.Should().ContainSingle<HttpHeaderNode>().Which
+                .ChildNodes.Should().ContainSingle<HttpHeaderSeparatorNode>();
+        }
+
+        [Fact]
+        public void body_separator_is_present()
+        {
+            var result = HttpRequestParser.Parse(
+            """
+            POST https://example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            Authorization: token xxx
+
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);         
+
+            var requestNode = result.SyntaxTree.RootNode
+                    .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+            requestNode.BodySeparatorNode.TextWithTrivia.Should().Be("\r\n");          
```

Comment:
Again, I think we should validate the token kind as opposed to the actual text for newlines. This would also address the test failure on Linux.